src/lib.rs

```diff
@@ -189,148 +247,56 @@ pub trait LogLevel {
     }
 }
 
-/// Default to [`log::Level::Error`]
-#[allow(clippy::exhaustive_structs)]
-#[derive(Copy, Clone, Debug, Default)]
-pub struct ErrorLevel;
-
-impl LogLevel for ErrorLevel {
-    fn default() -> Option<Level> {
-        Some(Level::Error)
-    }
+macro_rules! def_log_levels {
```

Comment:
This saves about 40 lines of repetitive code. It's not strictly necessary, but I think that's a reasonable thing.

Reply:
My bar for the use of Macros is high and I don't think this meets it.

Reply:
My threshold for this is a bit lower than yours. IMO, if a macro save a reasonable amount of code that I'll have to read / maintain without adding too much complexity then it's worth including. This is because they make things that look similar actually be similar. Usually that's somewhere around 5 repetitions of 3 things (e.g. struct definition, associated derives, implementing a trait hits my threshold pretty niicely).

I'm happy to go with your threshold here, though I'd suggest giving some future thought about whether that could do with a slight adjustment. If not, no big deal :)